deepsensor/model/model.py

```diff
@@ -648,6 +648,597 @@ def unnormalise_pred_array(arr, **kwargs):
+
+        # Perform patchwise predictions
+        preds = []
+        for task in tasks:
+            bbox = task["bbox"]
+
+            if bbox is None:
+                raise AttributeError(
+                    "For patchwise prediction, only tasks generated using a patch_strategy of 'sliding' are valid. \
+                        This task has a bbox value of None, indicating that it was generated with a patch_strategy of \
+                            'random' or None."
+                )
+
+            # Unnormalise coordinates of bounding box of patch
+            x1 = xr.DataArray([bbox[0], bbox[1]], dims="x1", name="x1")
+            x2 = xr.DataArray([bbox[2], bbox[3]], dims="x2", name="x2")
+            bbox_norm = xr.Dataset(coords={"x1": x1, "x2": x2})
+            bbox_unnorm = data_processor.unnormalise(bbox_norm)
+            unnorm_bbox_x1 = (
+                bbox_unnorm[orig_x1_name].values.min(),
+                bbox_unnorm[orig_x1_name].values.max(),
+            )
+            unnorm_bbox_x2 = (
+                bbox_unnorm[orig_x2_name].values.min(),
+                bbox_unnorm[orig_x2_name].values.max(),
+            )
+
+            # Determine X_t for patch, however, cannot assume min/max ordering of slice coordinates
+            # Check the order of coordinates in X_t, sometimes they are increasing or decreasing in order.
+            x1_coords = X_t.coords[orig_x1_name].values
+            x2_coords = X_t.coords[orig_x2_name].values
+
+            if x1_coords[0] < x1_coords[-1]:
+                x1_slice = slice(unnorm_bbox_x1[0], unnorm_bbox_x1[1])
+                x1_ascending = True
+            else:
+                x1_slice = slice(unnorm_bbox_x1[1], unnorm_bbox_x1[0])
+                x1_ascending = False
+
+            if x2_coords[0] < x2_coords[-1]:
+                x2_slice = slice(unnorm_bbox_x2[0], unnorm_bbox_x2[1])
+                x2_ascending = True
+            else:
+                x2_slice = slice(unnorm_bbox_x2[1], unnorm_bbox_x2[0])
+                x2_ascending = False
+
+            # Determine X_t for patch with correct slice direction
+            task_X_t = X_t.sel(**{orig_x1_name: x1_slice, orig_x2_name: x2_slice})
+
+            # Patchwise prediction
+            pred = self.predict(task, task_X_t)
+            # Append patchwise DeepSensor prediction object to list
+            preds.append(pred)
+
+        overlap_norm = tuple(
+            patch - stride for patch, stride in zip(patch_size, stride)
+        )
+        patch_overlap_unnorm = get_patch_overlap(
+            overlap_norm, data_processor, X_t, x1_ascending, x2_ascending
+        )
+
+        patches_per_row = get_patches_per_row(preds)
+        stitched_prediction = stitch_clipped_predictions(
+            preds, patch_overlap_unnorm, patches_per_row, x1_ascending, x2_ascending
+        )
+
+        ## Cast prediction into DeepSensor.Prediction object.
+        # TODO make this into seperate method.
```

Comment:
Would you like to do this in this PR, and add it to deepsensor.model.pred?

Reply:
I now have a better understanding of what Martin's code is doing here to create a `Prediction` object, i.e. copying one of the patch predictions and extending it. But do you think it would be better to instantiate a new `Prediction` directly? I'm not sure whether I should be using `Prediction.assign()` - what do you think?

Reply:
`pred.assign` isn't really what you want for stitching predictions together because that method just assigns data in bulk to the xarray or pandas objects.

TBH I've stared at this for a while and I'm not sure what you mean by 'copying one of the patch predictions and extending it' or where I should be looking, partly because it's difficult to follow `predict_patchwise` because of all the nested functions. In particular I don't understand what's being done with `prediction` below and why we can't just `return stitched_prediction` -- 

As typing I just realised that `stitch_clipped_predictions` is returning a dict not a `Prediction`, and the code below has a bunch of redundant lines and is simply overwriting the entries of a copy of the first patch `Prediction` with the xarray objects in `stitched_predictions`. I think you can simplify and resolve this if you resolve my comment above about making `stitch_clipped_predictions` return a `Prediction` directly.